pkg/restore/restore.go

```diff
@@ -1035,6 +1035,37 @@ func (ctx *restoreContext) restoreItem(obj *unstructured.Unstructured, groupReso
 		default:
 			ctx.log.Infof("Restoring persistent volume as-is because it doesn't have a snapshot and its reclaim policy is not Delete.")
 
+			shouldRenamePV, err := shouldRenamePV(ctx, obj, resourceClient)
```

Comment:
Hmm, since this is the same logic as the first case in the `switch`, I'm wondering if we should consolidate this into a function, or maybe do some of the logic before the `switch`.

The downside of moving it before is that it may be more work even in cases where there is no mapping to take place.

Reply:
@nrb So I looked into this, but it looks like it would be difficult to drop all of the logic here into one function. Basically the logic added to the default case does 3 things:
1) Check whether we should rename the PV
2) Rename the PV if 1) is true
3) Remap the namespace if we need to

So 1+2 and 3 are somewhat independent of each other. Currently 2) and 3) are done in a different order in the two code paths, but there's really no need for that. However, the part that makes putting all of this code into a func is the fact that in the `hasSnapshot` case, there is some logic required between 1) and 2) which is not there in the default case. One thing I could consolidate is the block of code *inside* the `if shouldRenamePV` block -- I could put that into a a new `func renamePV(...)` which would be called inside the if block in both cases, if that would be enough of an improvement in terms of avoiding duplication.